media-autobuild_suite.bat

```diff
@@ -1337,49 +1337,39 @@ if not exist %instdir%\mintty.lnk (
     echo -------------------------------------------------------------------------------
     echo.- make a first run
     echo -------------------------------------------------------------------------------
-    call :runBash firstrun.log "-c exit"
+    call :runBash firstrun.log exit
+
+    sed -i "s/#Color/Color/;s/^^IgnorePkg.*/#&/" %instdir%\%msys2%\etc\pacman.conf
 
     echo.-------------------------------------------------------------------------------
     echo.first update
     echo.-------------------------------------------------------------------------------
-    (
-        echo.echo -ne "\033]0;first msys2 update\007"
-        echo.pacman --noconfirm -Sy --asdeps pacman-mirrors
-        echo.sed -i "s;^^IgnorePkg.*;#&;" /etc/pacman.conf
-        echo.sleep ^4
-        echo.exit
-    )>%build%\firstUpdate.sh
-    call :runBash firstUpdate.log "%build%\firstUpdate.sh"
-    del %build%\firstUpdate.sh
+    title first msys2 update
+    call :runBash firstUpdate.log pacman --noconfirm -Sy --asdeps pacman-mirrors
 
     echo.-------------------------------------------------------------------------------
     echo.critical updates
     echo.-------------------------------------------------------------------------------
-    %bash% -lc "pacman -S --needed --ask=20 --noconfirm --asdeps bash pacman msys2-runtime"
+    pacman -S --needed --ask=20 --noconfirm --asdeps bash pacman msys2-runtime
```

Comment:
Isn't the `call :runBash` missing here? Does raw calling pacman work from batch?

Reply:
After extensive testing on using pacman directly from batch, cmd, and powershell, pacman seems to work fine outside of bash.

A long time ago, I did have some problems with pacman complaining about missing commands, but I think that was because I called pacman by full path without having the msys2 folders in PATH

we could use runBash with this, but since it didn't have a log file named nor using mintty previously, I didn't. Plus since it's updating bash, pacman, and msys2-runtime, I think it will ask to close without calling exit and Idk if there's a way to easily make it close without user interaction

Reply:
If it closes on its own and doesn't require closing the .bat so that pacman.exe is updated, should be fine.